File: 231. Power of Two/IsPowerOfTwo.py
'''
https://leetcode.com/problems/power-of-two/
'''


class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        
        while n > 2:
            if n % 2 != 0:  return False
            n //=2
        
        return True if n > 0 else False

File: 231. Power of Two/is_power_of_two.rs
impl Solution {
    
    // O(logn) time,
    // O(logn) space,
    // Approach: recursion, 
    pub fn is_power_of_two(n: i32) -> bool {
        if n == 1 {
            return true;
        }
        
        if n <= 0 {
            return false;
        }        
        
        match n % 2 {
            1 => false,
            _ => Solution::is_power_of_two(n/2),
        }
    }
}